libsolidity/interface/CompilerStack.cpp

```diff
@@ -649,55 +674,85 @@ bool CompilerStack::compile(State _stopAfter)
 	// Only compile contracts individually which have been requested.
 	map<ContractDefinition const*, shared_ptr<Compiler const>> otherCompilers;
 
-	for (Source const* source: m_sourceOrder)
-		for (ASTPointer<ASTNode> const& node: source->ast->nodes())
-			if (auto contract = dynamic_cast<ContractDefinition const*>(node.get()))
-				if (isRequestedContract(*contract))
-				{
-					try
+	if (!m_evmAssemblyJson.empty())
+	{
+		solAssert(m_importedSources, "");
+		solAssert(m_evmAssemblyJson.size() == 1, "");
+
+		string const evmSourceName = m_evmAssemblyJson.begin()->first;
+		Json::Value const evmJson = m_evmAssemblyJson.begin()->second;
+
+		evmasm::Assembly::OptimiserSettings optimiserSettings;
+		optimiserSettings.evmVersion = m_evmVersion;
+		optimiserSettings.expectedExecutionsPerDeployment = m_optimiserSettings.expectedExecutionsPerDeployment;
+		optimiserSettings.runCSE = m_optimiserSettings.runCSE;
```

Comment:
Don't we have a translation function here?

Reply:
Thats true! I saw that once too, but then I was not able to find it anymore. However, found it now `CompilerContext::translateOptimiserSettings` - refactored it - so there is now a static function `evmasm::Assembly::OptimiserSettings::translateSettings`.

Reply:
I originally did it like that in #10286, but back then @axic objected that it would introduce unnecessary cross dependencies (https://github.com/ethereum/solidity/pull/10286#issuecomment-827969786):

> If we are not doing it, I'd prefer the original version where we just copied the settings within libyul, without creating this weird set off cross dependencies between libevmasm <-> libyul <-> libsolidity.

Reply:
Hmm.. I can see that the dependency between `libevmasm <-> libsolidity` was already introduced by https://github.com/ethereum/solidity/pull/11550. I saw that you where asking for feedback from @axic in that PR. However, it got merged like this. Thats why I would argue that it is ok to have these dependencies - because they are already there.

Looking at the current dependencies again: `libsolidity/codegen/CompilerContext.cpp` is now using the translate function defined in "libevmasm/Assembly.cpp". So we have a dependency between `libsolidity->libevmasm`. This function uses a type `frontend::OptimiserSettings` from `libsolidity` - but we also use `evmasm::AssemblyItems` here. That means we have now `libsolidity <-> libevmasm`. `libsolidity/codegen/CompilerContext.cpp` also uses other `libevmasm` types like `evmasm::AssemblyItem`. Looking at `libyul/AssemblyStack.cpp` again, that was now renamed to `libyul/YulStack.cpp`. It is also having the dependency to `libevmasm`, because this dependency is already defined by the copied "translateOptimiserSettings" function - but this dependency is already there, because of the usage of e.g. `evmasm::LinkerObject` or `evmasm::Assembly`. So we have already a strong interdependence between these module. 

So the these cross dependencies where are already in place.

I think it would be good to replace the code replication in `libyul/YulStack.cpp`, so that `YulStack::assembleEVMWithDeployed` will now also use `Assembly::OptimiserSettings::translateSettings`.

Reply:
After talking to @cameel, we decided to just remove the translation function from this PR. I will create another PR that will only introduce that translation function.

Reply:
See https://github.com/ethereum/solidity/pull/13405